### ISSUE TITLE
Bug: eslint --env-info doesn't report correct Node/eslint version

### ISSUE THREAD

**Issue by markboyall** [eslint/eslint#19848]
### Environment

Correct: 
Node version: v24.2.0
npm version: v8.5.5
Local ESLint version: 9.28.0
Global ESLint version: Not found
Operating System: win32 10.0.19045

Wrong:
Node version: v16.15.0
npm version: v8.5.5
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: win32 10.0.19045

### What parser are you using?

@typescript-eslint/parser

### What did you do?

See the reproduction repository, have some version of Node installed that is not 24.2.0, but instead of the readme command which reproduces a different issue, just run ./node_modules/node/bin/node.exe ./node_modules/eslint/bin/eslint.js --env-info.

### What did you expect to happen?

The Node version should be the Node version that is actually being run. The eslint version should obviously be the current eslint version.

### What actually happened?

eslint reports the installed Node, not the running Node. Additionally, eslint fails to report its own version.

### Link to Minimal Reproducible Example

https://github.com/markboyall/eslint-file-included

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
How are you running ESLint? Please provide the actual command.

**Comment by markboyall** [comment]
./node_modules/node/bin/node.exe ./node_modules/eslint/bin/eslint.js --env-info as stated.

![Image](https://github.com/user-attachments/assets/a1764ea9-c1e5-4079-9693-dc61a3ed8818)

**Comment by nzakas** [comment]
Thanks, I can reproduce this. Very strange.

FWIW, you can run ESLint more simply via `npx eslint --env-info`.

**Comment by nzakas** [comment]
This is interesting:

```shell
nzaka@X1-Laptop MINGW64 ~/projects/eslint/tmp/eslint-file-included (master)
$ node node_modules/eslint/bin/eslint.js --env-info
Environment Info:

Node version: v22.15.0
npm version: v11.0.0
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: win32 10.0.26100

nzaka@X1-Laptop MINGW64 ~/projects/eslint/tmp/eslint-file-included (master)
$ npx eslint --env-info
Environment Info:

Node version: v24.2.0
npm version: v11.0.0
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: win32 10.0.26100
```

**Comment by markboyall** [comment]
I'm avoiding any use of environmental npm or node during the build due to needing to build old versions and Node not always being backwards or forwards compatible, so we're using node as an npm package and controlling it directly to make sure builds of older versions are using the versions of Node they were built for. Thanks for looking into this issue

**Comment by nzakas** [comment]
Here's the problem, you have this package installed:
https://www.npmjs.com/package/node

This creates an entry for `node_modules/.bin/node`, so when we run `node --version` inside of ESLint, it actually runs that command instead of Node.js. You'll note the version of that package is 24.2.0, the exact version that's output with `--env-info`.

You may want to double-check why you have that package installed as I'm willing to bet this isn't the only issue you'll run into with that binary sitting in your `node_modules`.

On our side, I think we can switch to using `process.version` instead of running `node --version`. I'm not sure why we did that but it's been there since this functionality was introduced so I think it's safe to switch.

**Comment by TKDev7** [comment]
I'll work on this.
